web/src/components/overview/NetworkSection.jsx

```diff
@@ -23,35 +23,81 @@ import React, { useEffect, useState } from "react";
 import { sprintf } from "sprintf-js";
 
 import { Em, Section, SectionSkeleton } from "~/components/core";
-import { ConnectionTypes } from "~/client/network";
 import { useInstallerClient } from "~/context/installer";
+import { NetworkEventTypes } from "~/client/network";
 import { formatIp } from "~/client/network/utils";
 import { _, n_ } from "~/i18n";
 
 export default function NetworkSection() {
   const { network: client } = useInstallerClient();
-  const [connections, setConnections] = useState(undefined);
+  const [devices, setDevices] = useState(undefined);
 
   useEffect(() => {
-    if (connections !== undefined) return;
+    return client.onNetworkChange(({ type, payload }) => {
+      switch (type) {
+        case NetworkEventTypes.DEVICE_ADDED: {
+          setDevices((devs) => {
+            const newDevices = devs.filter((d) => d.name !== payload.name);
+            if (payload.connection) return newDevices;
 
-    client.connections().then(setConnections);
-  }, [client, connections]);
+            return [...newDevices, client.fromApiDevice(payload)];
+          });
+          break;
+        }
 
+        case NetworkEventTypes.DEVICE_UPDATED: {
+          const [name, data] = payload;
+          setDevices(devs => {
+            const newDevices = devs.filter((d) => d.name !== name);
+            return [...newDevices, client.fromApiDevice(data)];
+          });
+          break;
+        }
+
+        case NetworkEventTypes.DEVICE_REMOVED: {
+          setDevices(devs => devs.filter((d) => d.name !== payload));
+          break;
+        }
+      }
+    });
+  }, [client, devices]);
+
+  useEffect(() => {
+    if (devices !== undefined) return;
+
+    client.devices().then(setDevices);
+  }, [client, devices]);
+
+  const nameFor = (device) => {
+    if (device.connection === undefined || device.connection.trim() === "") return device.name;
+
+    return device.connection;
```

Comment:
UX nitpicking...
My slightly unusual use case is not happy with this UX choice :smiling_face_with_tear: 

This line is responsible for the "Sarambeque" label in the "Overview with two devices" screenshot. I guess it is nice to see the ESSID right away, but it becomes confusing in my own use case:

I have a built-in WiFi interface and a USB WiFi dongle to get better signal strength. NetworkManager GUI simply says "Connected to MyEssid" but I need to click in the detailed view to check _which_ interface is connected to it. Keeping the interface name (wlan0 connected to MyEssid) would make my life easier.